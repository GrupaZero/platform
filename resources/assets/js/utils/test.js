import Translations from '../lang/locales.js';
import VueI18n from 'vue-i18n';
import {mount} from 'avoriaz';

function vueMount(component, arg) {
    component.i18n = new VueI18n({
        locale: 'en',
        fallbackLocale: 'en',
        messages: Translations
    });
    return mount(component, arg)
}

export {vueMount};