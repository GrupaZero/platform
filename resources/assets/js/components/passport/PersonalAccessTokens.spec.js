import test from 'ava'
import {vueMount} from '../../utils/test'
import Bluebird from 'bluebird'
import PersonalAccessTokens from './PersonalAccessTokens.vue'

test.beforeEach(() => {
    axiosMock.reset()
    axiosMock.delayResponse = 0
})

test.serial('Init state without personal access tokens', t => {
    t.plan(1)

    axiosMock.onGet('/oauth/scopes').reply(200, [])
    axiosMock.onGet('/oauth/personal-access-tokens').reply(200, [])

    const wrapper = vueMount(PersonalAccessTokens)

    return Bluebird
    .delay()
    .then(() => {
        t.snapshot(wrapper.html())
    })

})

test.serial('Should render OAuth personal access tokens table', t => {
    t.plan(3)

    axiosMock.onGet('/oauth/scopes')
    .reply(200, [
        {id: 1},
        {id: 2}
    ])

    axiosMock.onGet('/oauth/personal-access-tokens')
    .reply(200, [
        {id: 1, name: 'OAuth Token 1'},
        {id: 2, name: 'OAuth Token 2'}
    ])

    const wrapper = vueMount(PersonalAccessTokens)
    t.is(0, wrapper.find('tr').length) // Before rendering

    return Bluebird
    .delay()
    .then(() => {
        t.snapshot(wrapper.html())
        t.is(3, wrapper.find('tr').length) // 2 tokens + header
    })

})

test.serial('Should render OAuth personal access tokens table after 100 ms delay', t => {
    t.plan(3)

    axiosMock.delayResponse = 100

    axiosMock.onGet('/oauth/scopes')
    .reply(200, [
        {id: 1},
        {id: 2}
    ])

    axiosMock.onGet('/oauth/personal-access-tokens')
    .reply(200, [
        {id: 1, name: 'OAuth Token 1'},
        {id: 2, name: 'OAuth Token 2'}
    ])

    const wrapper = vueMount(PersonalAccessTokens)
    t.is(0, wrapper.find('tr').length) // Before rendering

    return Bluebird
    .delay(105)
    .then(() => {
        t.snapshot(wrapper.html())
        t.is(3, wrapper.find('tr').length) // 2 tokens + header
    })

})
