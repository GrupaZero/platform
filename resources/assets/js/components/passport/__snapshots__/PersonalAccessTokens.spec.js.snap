// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init state without personal access tokens 1`] = `
"<div><div><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\"><div style=\\"display: flex;\\"><span>
                            Personal Access Tokens
                        </span><a class=\\"action-link\\">
                            Create New Token
                        </a></div></div><div class=\\"panel-body\\"><p class=\\"m-b-none\\">
                        You have not created any personal access tokens.
                    </p><!----></div></div></div><div id=\\"modal-create-token\\" tabindex=\\"-1\\" role=\\"dialog\\" class=\\"modal fade\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><button type=\\"button \\" data-dismiss=\\"modal\\" aria-hidden=\\"true\\" class=\\"close\\">×</button><h4 class=\\"modal-title\\">
                            Create Token
                        </h4></div><div class=\\"modal-body\\"><!----><form role=\\"form\\" class=\\"form-horizontal\\"><div class=\\"form-group\\"><label class=\\"col-md-4 control-label\\">Name</label><div class=\\"col-md-6\\"><input id=\\"create-token-name\\" type=\\"text\\" name=\\"name\\" class=\\"form-control\\"></div></div><!----></form></div><div class=\\"modal-footer\\"><button type=\\"button\\" data-dismiss=\\"modal\\" class=\\"btn btn-default\\">Close</button><button type=\\"button\\" class=\\"btn btn-primary\\">
                            Create
                        </button></div></div></div></div><div id=\\"modal-access-token\\" tabindex=\\"-1\\" role=\\"dialog\\" class=\\"modal fade\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><button type=\\"button\\" data-dismiss=\\"modal\\" aria-hidden=\\"true\\" class=\\"close\\">×</button><h4 class=\\"modal-title\\">
                            Personal Access Tokens
                        </h4></div><div class=\\"modal-body\\"><p>
                            Here is your new personal access token. This is the only time it will be shown so don't lose it! You may now use this token to make API requests.
                        </p><pre><code></code></pre></div><div class=\\"modal-footer\\"><button type=\\"button\\" data-dismiss=\\"modal\\" class=\\"btn btn-default\\">Close</button></div></div></div></div></div>"
`;

exports[`Should render OAuth personal access tokens table 1`] = `
"<div><div><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\"><div style=\\"display: flex;\\"><span>
                            Personal Access Tokens
                        </span><a class=\\"action-link\\">
                            Create New Token
                        </a></div></div><div class=\\"panel-body\\"><!----><table class=\\"table table-borderless m-b-none\\"><thead><tr><th>Name</th><th></th></tr></thead><tbody><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 1
                                </td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Delete
                                    </a></td></tr><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 2
                                </td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Delete
                                    </a></td></tr></tbody></table></div></div></div><div id=\\"modal-create-token\\" tabindex=\\"-1\\" role=\\"dialog\\" class=\\"modal fade\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><button type=\\"button \\" data-dismiss=\\"modal\\" aria-hidden=\\"true\\" class=\\"close\\">×</button><h4 class=\\"modal-title\\">
                            Create Token
                        </h4></div><div class=\\"modal-body\\"><!----><form role=\\"form\\" class=\\"form-horizontal\\"><div class=\\"form-group\\"><label class=\\"col-md-4 control-label\\">Name</label><div class=\\"col-md-6\\"><input id=\\"create-token-name\\" type=\\"text\\" name=\\"name\\" class=\\"form-control\\"></div></div><div class=\\"form-group\\"><label class=\\"col-md-4 control-label\\">Scopes</label><div class=\\"col-md-6\\"><div><div class=\\"checkbox\\"><label><input type=\\"checkbox\\">

                                                    1
                                            </label></div></div><div><div class=\\"checkbox\\"><label><input type=\\"checkbox\\">

                                                    2
                                            </label></div></div></div></div></form></div><div class=\\"modal-footer\\"><button type=\\"button\\" data-dismiss=\\"modal\\" class=\\"btn btn-default\\">Close</button><button type=\\"button\\" class=\\"btn btn-primary\\">
                            Create
                        </button></div></div></div></div><div id=\\"modal-access-token\\" tabindex=\\"-1\\" role=\\"dialog\\" class=\\"modal fade\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><button type=\\"button\\" data-dismiss=\\"modal\\" aria-hidden=\\"true\\" class=\\"close\\">×</button><h4 class=\\"modal-title\\">
                            Personal Access Tokens
                        </h4></div><div class=\\"modal-body\\"><p>
                            Here is your new personal access token. This is the only time it will be shown so don't lose it! You may now use this token to make API requests.
                        </p><pre><code></code></pre></div><div class=\\"modal-footer\\"><button type=\\"button\\" data-dismiss=\\"modal\\" class=\\"btn btn-default\\">Close</button></div></div></div></div></div>"
`;

exports[`Should render OAuth personal access tokens table after 100 ms delay 1`] = `
"<div><div><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\"><div style=\\"display: flex;\\"><span>
                            Personal Access Tokens
                        </span><a class=\\"action-link\\">
                            Create New Token
                        </a></div></div><div class=\\"panel-body\\"><!----><table class=\\"table table-borderless m-b-none\\"><thead><tr><th>Name</th><th></th></tr></thead><tbody><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 1
                                </td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Delete
                                    </a></td></tr><tr><td style=\\"vertical-align: middle;\\">
                                    OAuth Token 2
                                </td><td style=\\"vertical-align: middle;\\"><a class=\\"action-link text-danger\\">
                                        Delete
                                    </a></td></tr></tbody></table></div></div></div><div id=\\"modal-create-token\\" tabindex=\\"-1\\" role=\\"dialog\\" class=\\"modal fade\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><button type=\\"button \\" data-dismiss=\\"modal\\" aria-hidden=\\"true\\" class=\\"close\\">×</button><h4 class=\\"modal-title\\">
                            Create Token
                        </h4></div><div class=\\"modal-body\\"><!----><form role=\\"form\\" class=\\"form-horizontal\\"><div class=\\"form-group\\"><label class=\\"col-md-4 control-label\\">Name</label><div class=\\"col-md-6\\"><input id=\\"create-token-name\\" type=\\"text\\" name=\\"name\\" class=\\"form-control\\"></div></div><div class=\\"form-group\\"><label class=\\"col-md-4 control-label\\">Scopes</label><div class=\\"col-md-6\\"><div><div class=\\"checkbox\\"><label><input type=\\"checkbox\\">

                                                    1
                                            </label></div></div><div><div class=\\"checkbox\\"><label><input type=\\"checkbox\\">

                                                    2
                                            </label></div></div></div></div></form></div><div class=\\"modal-footer\\"><button type=\\"button\\" data-dismiss=\\"modal\\" class=\\"btn btn-default\\">Close</button><button type=\\"button\\" class=\\"btn btn-primary\\">
                            Create
                        </button></div></div></div></div><div id=\\"modal-access-token\\" tabindex=\\"-1\\" role=\\"dialog\\" class=\\"modal fade\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><button type=\\"button\\" data-dismiss=\\"modal\\" aria-hidden=\\"true\\" class=\\"close\\">×</button><h4 class=\\"modal-title\\">
                            Personal Access Tokens
                        </h4></div><div class=\\"modal-body\\"><p>
                            Here is your new personal access token. This is the only time it will be shown so don't lose it! You may now use this token to make API requests.
                        </p><pre><code></code></pre></div><div class=\\"modal-footer\\"><button type=\\"button\\" data-dismiss=\\"modal\\" class=\\"btn btn-default\\">Close</button></div></div></div></div></div>"
`;
